src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -610,11 +605,6 @@ void enforceNotNullWhenForPendingReturn(PendingBranch pendingReturn, BoundExpres
                         enforceParameterNotNullWhen(returnStatement.Syntax, parameters, sense: true, stateWhen: pendingReturn.StateWhenTrue);
                         enforceParameterNotNullWhen(returnStatement.Syntax, parameters, sense: false, stateWhen: pendingReturn.StateWhenFalse);
                     }
-                    else
-                    {
-                        enforceParameterNotNullWhen(returnStatement.Syntax, parameters, sense: true, stateWhen: pendingReturn.State);
```

Comment:
So this works because we enter a split state after visiting a `bool` constant, right?

Reply:
Discussed offline.